lib/coverage.js

```diff
@@ -385,6 +385,50 @@ internals.instrument = function (filename) {
 };
 
 
+internals.traverse = function (coveragePath, options) {
+
+    let nextPath = null;
+    const traverse = function (path) {
+
+        let files = [];
+        nextPath = path;
+
+        const pathStat = Fs.statSync(path);
+        if (pathStat.isFile()) {
+            return path;
+        }
+
+        Fs.readdirSync(path).forEach((filename) => {
+
+            nextPath = Path.join(path, filename);
+            const stat = Fs.statSync(nextPath);
+            if (stat.isDirectory() &&
+                !options['coverage-flat']) {
+
+                files = files.concat(traverse(nextPath, options));
+                return;
+            }
+
+            if (stat.isFile() &&
+                Path.basename(nextPath)[0] !== '.') {
```

Comment:
In the `test/coverage/coverage-all/nested-folder/` you can try adding a `.dir/somefile.js` and verify that this case is tested

Reply:
The coverage path was incorrect to fully test this scenario, updated!